test/e2e/sidecar_namespace_test.go

```diff
@@ -0,0 +1,106 @@
+// +build smoke
+
+package e2e
+
+import (
+	goctx "context"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/inject"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type SidecarNamespaceTestSuite struct {
+	suite.Suite
+}
+
+func (suite *SidecarNamespaceTestSuite) SetupSuite() {
+	t = suite.T()
+	var err error
+	ctx, err = prepare(t)
+	if err != nil {
+		if ctx != nil {
+			ctx.Cleanup()
+		}
+		require.FailNow(t, "Failed in prepare")
+	}
+	fw = framework.Global
+	namespace, _ = ctx.GetNamespace()
+	require.NotNil(t, namespace, "GetNamespace failed")
+
+	addToFrameworkSchemeForSmokeTests(t)
+}
+
+func (suite *SidecarNamespaceTestSuite) TearDownSuite() {
+	handleSuiteTearDown()
+}
+
+func TestSidecarNamespaceSuite(t *testing.T) {
+	suite.Run(t, new(SidecarNamespaceTestSuite))
+}
+
+func (suite *SidecarNamespaceTestSuite) SetupTest() {
+	t = suite.T()
+}
+
+func (suite *SidecarNamespaceTestSuite) AfterTest(suiteName, testName string) {
+	handleTestFailure()
+}
+
+// Sidecar runs a test with the agent as sidecar
+func (suite *SidecarNamespaceTestSuite) TestSidecarNamespace() {
+	cleanupOptions := &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval}
+
+	jaegerInstanceName := "agent-as-sidecar-namespace"
+	j := getJaegerAgentAsSidecarDefinition(jaegerInstanceName, namespace)
+	err := fw.Client.Create(goctx.TODO(), j, cleanupOptions)
+	require.NoError(t, err, "Failed to create jaeger instance")
+	defer undeployJaegerInstance(j)
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for Jaeger instance deployment")
+
+	dep := getVertxDefinition(map[string]string{})
+	err = fw.Client.Create(goctx.TODO(), dep, cleanupOptions)
```

Comment:
Could the test perhaps wait for the deployment to stabilize and check that no sidecar exists? This way, it asserts that the sidecar was injected because of the namespace annotation.